homeassistant/components/fritz/common.py

```diff
@@ -175,23 +185,14 @@ def setup(self) -> None:
 
         self._update_available, self._latest_firmware = self._update_device_info()
 
-    async def async_start(self, options: MappingProxyType[str, Any]) -> None:
-        """Start FritzHosts connection."""
-        self.fritz_hosts = FritzHosts(fc=self.connection)
-        self._options = options
-        await self.hass.async_add_executor_job(self.scan_devices)
-
-        self._cancel_scan = async_track_time_interval(
-            self.hass, self.scan_devices, timedelta(seconds=TRACKER_SCAN_INTERVAL)
-        )
-
     @callback
-    def async_unload(self) -> None:
-        """Unload FritzboxTools class."""
-        _LOGGER.debug("Unloading FRITZ!Box router integration")
-        if self._cancel_scan is not None:
-            self._cancel_scan()
-            self._cancel_scan = None
+    async def _async_update_data(self) -> None:
```

Comment:
Are we guaranteed to have at least one entity added when setting up the integration? The coordinator won't update automatically unless there are listeners (entities) added. So unless there's at least one entity added during config entry setup, no new scans will be done and no new entities added for new devices.